solidity/contracts/token/extensions/OPL2ToL1TokenBridgeNative.sol

```diff
@@ -63,97 +74,128 @@ contract OpL2NativeTokenBridge is HypNative {
+     * - Finalize message with the actual amount to finalize the withdrawal
+     * transferRemote typically has the dispatch of the message as the 4th and final step. However, in this case we want the Hyperlane messageId to be passed via the rollup bridge.
+     */
+    function transferRemote(
         uint32 _destination,
         bytes32 _recipient,
-        uint256 _amount,
-        bytes memory _hookMetadata,
-        address _hook
-    ) internal virtual override returns (bytes32) {
+        uint256 _amount
+    ) public payable virtual override returns (bytes32) {
+        // 1. No external fee calculation necessary
         require(
             _amount > 0,
             "OP L2 token bridge: amount must be greater than 0"
         );
 
-        // refund first message fees to address(this) to cover second message
-        bytes32 proveMessageId = super._transferRemote(
+        // 2. Prepare the "dispatch" of messages by actually dispatching the Hyperlane messages
+
+        // Dispatch proof message (no token amount)
+        bytes32 proveMessageId = _Router_dispatch(
             _destination,
-            _recipient,
-            0,
+            msg.value - _amount,
+            TokenMessage.format(_recipient, 0),
             _proveHookMetadata(),
-            _hook
+            address(hook)
         );
 
-        bytes32 withdrawMessageId = super._transferRemote(
+        // Dispatch withdrawal message (token + fee)
+        bytes32 withdrawMessageId = _Router_dispatch(
             _destination,
-            _recipient,
-            _amount,
+            address(this).balance - _amount,
+            TokenMessage.format(_recipient, _amount),
             _finalizeHookMetadata(),
-            _hook
+            address(hook)
         );
 
         // include for legible error message
-        _transferFromSender(_amount);
+        require(
+            address(this).balance >= _amount,
+            "OP L2 token bridge: insufficient balance"
+        );
+
+        // 3. Emit event manually
+        emit SentTransferRemote(_destination, _recipient, _amount);
 
         // used for mapping withdrawal to hyperlane prove and finalize messages
         bytes memory extraData = OPL2ToL1Withdrawal.encodeData(
             proveMessageId,
             withdrawMessageId
         );
+
+        // 4. "Dispatch" the message by calling the L2 bridge to transfer native tokens
         l2Bridge.bridgeETHTo{value: _amount}(
             _recipient.bytes32ToAddress(),
             OP_MIN_GAS_LIMIT_ON_L1,
             extraData
         );
 
         if (address(this).balance > 0) {
-            address refundAddress = _hookMetadata.getRefundAddress(msg.sender);
-            require(
-                refundAddress != address(0),
-                "OP L2 token bridge: refund address is 0"
-            );
-            payable(refundAddress).sendValue(address(this).balance);
+            payable(msg.sender).sendValue(address(this).balance);
```

Comment:
this seems wrong

Reply:
we dont want any balance on this contract, so I think its correct to send the whole balance
I meant we aren't using the metadata specified refund address

Reply:
I guess we don't have access to the metadata here... hmm